backend/infrahub/services/adapters/event/__init__.py

```diff
@@ -1,41 +1,32 @@
 from __future__ import annotations
 
 import asyncio
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Optional
 
 from prefect.events import emit_event
 
-from infrahub.exceptions import InitializationError
-
 if TYPE_CHECKING:
     from infrahub.events import InfrahubEvent
-    from infrahub.services import InfrahubServices
+    from infrahub.services import InfrahubMessageBus
 
 
 class InfrahubEventService:
     """Base class for infrahub event service"""
 
-    def __init__(self) -> None:
-        self._service: InfrahubServices | None = None
-
-    @property
-    def service(self) -> InfrahubServices:
-        if not self._service:
-            raise InitializationError("Event is not initialized with a service")
-
-        return self._service
-
-    async def initialize(self, service: InfrahubServices) -> None:
-        """Initialize the event service"""
-        self._service = service
+    def __init__(self, message_bus: Optional[InfrahubMessageBus] = None) -> None:
```

Comment:
InfrahubServices | None = None
```suggestion
    def __init__(self, message_bus: InfrahubServices | None = None) -> None:
```